### PR TITLE
add test and integration steps + test results

### PR DESCRIPTION
**Pull request by prunelsigma**
- ci separate into steps, fix the `test_results` output, cache the go modules
- fix the async test in `getconnectedusers`

### CODE REVIEW

File: internal/user/usercontroller/user_controller_test.go [hunk 1]
```diff
@@ -334,17 +334,25 @@ func TestGetConnectedUsers_ReturnsConnectedUsers(t *testing.T) {
...
-	userInfo2 := &usercontroller.UserInfo{Name: modelUser2.Name}
-
-	mockStream3.EXPECT().Send(userInfo1).Times(1)
-	mockStream3.EXPECT().Send(userInfo2).Times(1)
+	expectedUsers := make(map[string]usercontroller.UserInfo)
+	expectedUsers[modelUser1.Name] = usercontroller.UserInfo{Name: modelUser1.Name}
+	expectedUsers[modelUser2.Name] = usercontroller.UserInfo{Name: modelUser2.Name}
+
+	mockStream3.EXPECT().Send(gomock.Any()).DoAndReturn(
+		func(info *usercontroller.UserInfo) error {
```
Comment (Tibo-lg) [inline internal/user/usercontroller/user_controller_test.go:342]: I'm a bit picky, but I feel this wouldn't catch an issue if the stream returned two times the same username.

Comment (prunelsigma) [reply internal/user/usercontroller/user_controller_test.go:342]: even with the user found is removed from the map after ?

Comment (prunelsigma) [reply internal/user/usercontroller/user_controller_test.go:342]: and call only 2 times ?

Comment (Tibo-lg) [reply internal/user/usercontroller/user_controller_test.go:342]: Good point sorry I had missed the deletion, then I guess it works!

**Review by Tibo-lg — APPROVED** [review]
LGTM

**Pull request merged**